src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -369,6 +373,42 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [isMobile]);
 
+    const onPageResize = useCallback(() => {
+      const hasHorizontalScroll = document.documentElement.scrollWidth - document.documentElement.clientWidth > 0;
+      if (!hasHorizontalScroll) {
+        return;
+      }
+
+      if (navigationOpen) {
+        onNavigationToggle(false);
+        return;
+      }
+
+      const drawerToClose = drawersOpenQueue[drawersOpenQueue.length - 1];
+      if (activeDrawer && activeDrawer?.id === drawerToClose) {
+        onActiveDrawerChange(null);
+      } else if (activeGlobalDrawersIds.includes(drawerToClose)) {
+        onActiveGlobalDrawersChange(drawerToClose);
+      }
+    }, [
+      activeDrawer,
+      activeGlobalDrawersIds,
+      drawersOpenQueue,
+      navigationOpen,
+      onActiveDrawerChange,
+      onActiveGlobalDrawersChange,
+      onNavigationToggle,
+    ]);
+
+    useEffect(() => {
+      if (isMobile) {
+        return;
+      }
+      const throttledOnPageResize = throttle(onPageResize, 500);
+      window.addEventListener('resize', throttledOnPageResize);
```

Comment:
You might wonder why I didn't use useResizeObserver—the reason is that this approach is easier to mock and test

Reply:
Sounds like a potential for future issues. 

Also, this makes app layout depending on user interactions – open and then resize renders a different UI than opening on a small screen initially. We should not allow such deviations.

Reply:
What is your suggestion here?

Reply:
Do something similar to `computeLayout` function. The decision for drawers visibility should be a pure function too.